packages/blade/src/components/List/_decisions/decisions.md

```diff
@@ -0,0 +1,312 @@
+- [API](#api)
+  - [List Component](#list-component)
+  - [ListItem Component](#listitem-component)
+  - [ListItemLink Component](#listitemlink-component)
+  - [Sample Usage](#sample-usage)
+  - [Alternative APIs](#alternative-apis)
+    - [Hierarchial with `variant` as `'ordered' | 'ordered-filled' | 'unordered'`](#hierarchial-with-variant-as-ordered--ordered-filled--unordered)
+    - [Originally proposed API](#originally-proposed-api)
+    - [Hierarchial with separate OrderedList \& UnorderedList Components](#hierarchial-with-separate-orderedlist--unorderedlist-components)
+    - [Hierarchial without explicit nested `List` grouping](#hierarchial-without-explicit-nested-list-grouping)
+    - [Hierarchial with a `nested` prop](#hierarchial-with-a-nested-prop)
+    - [Non-Hierarchial with a `level` prop](#non-hierarchial-with-a-level-prop)
+- [Accessibility](#accessibility)
+- [Open Questions](#open-questions)
+- [References](#references)
+
+
+## Design
+- [Figma - List](https://www.figma.com/file/jubmQL9Z8V7881ayUD95ps/Blade---Payment-Light?node-id=23205%3A446859&t=dT849yOxgX5WHgby-4)
+
+### Unordered List
+#### Unordered List with 3 levels of nesting
+
+<img src="./unordered-list.png" width="380" />
+
+#### Unordered List with custom icons for bullets
+
+<img src="./unordered-list-icon.png" width="380" />
+
+#### Unordered List with inline Link in items
+
+<img src="./unordered-list-link.png" width="380" />
+
+### Ordered List
+#### Ordered List with 3 levels of nesting
+> Level 1 will be 1, 2, 3,...
+
+> Level 2 will be a, b, c,...
+
+> Level 3 will be i, ii, iii,...
+
+<img src="./ordered-list.png" width="380" />
+
+#### Ordered List with _filled background_ numbers
+> No nesting will be allowed on this
+
+<img src="./ordered-list-filled.png" width="380" />
+
+#### Ordered List with inline Link in items
+
+<img src="./ordered-list-link.png" width="380" />
+
+
+## API
+
+### List Component
+| Prop | Type | Default | Description | Required |
+|---|---|---|---|---|
+| children | `ListItem` | `undefined` | List items to render. | ✅ |
+| variant | `ordered`, `ordered-filled`, `unordered` | `unordered` | Sets the list as ordered or unordered. |  |
+| icon | `BladeIcon` | `undefined` | Icon to be used for the items in the list. This can only be set when `variant` is `unordered`. |  |
+| size | `small`, `medium` | `medium` | Size of the list items. |  |
+
+### ListItem Component
+| Prop | Type | Default | Description | Required |
+|---|---|---|---|---|
+| children | `string`, `ListItemLink`, `List` | `undefined` | String to render in the list item. You may also pass a `ListItemLink` to render an inline link. You may also pass a `List` component which contains its own `ListItem` components to create a nested list.| ✅ |
+| icon | `BladeIcon` | `undefined` | Icon to be used for the items in the list. This can only be set when `variant` is `unordered`. |  |
+
+### ListItemLink Component
```

Comment:
<!-- Suggestions propose improvements to the current subject. It's important to be explicit and clear on what is being suggested and why it is an improvement. Consider using patches and the blocking or non-blocking decorations to further communicate your intent.  -->
**suggestion:** ‏From docs perspective, there are a few other use cases that we need to support
1. ListItem with inline link and text
2. ListItem with text and an image
3. ListItem with arbitrary components (we have a callout component that is used sometimes)

[Reference](https://razorpay.com/docs/payments/dashboard/support#registered-razorpay-users)

To support such outlier use cases (I would assume non-documentation products don't have such use-cases), should we still allow the consumer to provide custom children (using some override key if required) and let them handle the consistency of size?

The override key would deter the majority of projects from providing custom implementations while be flexible enough for niche use cases like docs. 

Thoughts?

Reply:
> ListItem with inline link and text
That is already supported in the current API. We will expose a component `ListItemLink` which can be used inline with text

> ListItem with text and an image
> ListItem with arbitrary components (we have a callout component that is used sometimes)
We have scoped this out of List component for now until design adds support for it. @saurav12 mentioned that we don't have enough use-cases for this and docs design will also be relooking at how List is to be used. @saurav12 can add more to this

`override` key might be a deviation from our API design. If we do add support for custom children then we should allow it for every project as a valid use-case imo. 

I'll discuss the support for text, image and arbitrary components again with @saurav12 to make sure we're aligned on the scope of the current List.

Reply:
> ListItemLink which can be used inline with text

For this, from the examples I can see, the entire list item would be a link. However, there is a use case in docs where the list item has some link and text.

Reply:
>override key might be a deviation from our API design

Makes sense. However, can we consider directly allowing any children and letting the users handle the styling deviations? This is similar to how any arbitrary value is allowed in spacing for the layout component. If a consumer wants to override the default behaviour then it should be their responsibility to adhere to the design spec.

Reply:
> For this, from the examples I can see, the entire list item would be a link. However, there is a use case in docs where the list item has some link and text.

We would support inline text + link. I'll add an example on the doc as well, that's a miss

Eg)
```jsx
<ListItem>
    For more information <ListItemLink href=''>Click here</ListItemLink>
</ListItem>
```